conans/client/generators/__init__.py

```diff
@@ -88,7 +88,7 @@ def __getitem__(self, key):
 def write_generators(conanfile, path, output):
     """ produces auxiliary files, required to build a project or a package.
     """
-    for generator_name in conanfile.generators:
+    for generator_name in set(conanfile.generators):
```

Comment:
```suggestion
    for generator_name in sorted(set(conanfile.generators)):
```

Reply:
It shouldn't be needed, is there any reason why you should care about the order of generators (even for testing)?

Reply:
Maybe a ``attr("slow")`` test checks the output and it comes in a different order? Or some weird interaction between different generators that try to create the same file (``conan_common.cmake``) in the future, and that fail randomly depending on the order. Not a big thing, just a minor detail, 1 out of 10 of this kind of things come later to bite. So can be left as-is.

Reply:
If it fails once I'll fix that test, noone should expect generators to be in any specific order.